File: answer_1/alstamber_scala/5.scala
object main5{
  def main(args: Array[String]) = {
    val n = readInt()

    println(solve(n))
  }

  def solve(n:Int) = solve_sub(Range(1, n + 1).toList)

  def solve_sub(seq:List[Int]):String = {
    seq match {
      case Nil => ""
      case xs => five(xs.take(5)) + solve_sub(xs.drop(5))
    }
  }

  def five(seq:List[Int]):String = {
    seq match {
      case (head::Nil) => head.toString + "\n"
      case (head::tail) => head.toString + "," + five(tail)
    }
  }
}

File: answer_1/nomeaning_haskell/5.hs
five [x] = (show x) ++ ['\n']
five (a:x) =  (show a) ++ [','] ++ (five x)
rec [] = ""
rec x = (five (take 5 x)) ++ (rec $ drop 5 x)
solve n = rec [1..n]
main = interact $ solve . (read::String->Int)
